### PR TITLE
feat(react-tags): Add TagGroup with context

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
Add `TagGroup` component that wraps around Tag/TagButton, and providing context.
Currently there's only `size` in TagGroupContext

usage:
```tsx
<TagGroup size="small">
  <Tag>Tag 1</Tag>
  <TagButton>Tag 2</TagButton>
</TagGroup>
```

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                  |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                   Change |
| :--------------------------------------------------------------------------------- | --------------------------: | --------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr> | `20.637 kB`<br />`7.332 kB` | `20.916 kB`<br />`7.449 kB` | `279 B` <br />`117 B`  |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/36b914d0bbdad6bf5ee88204ec44f3a5a9b47936'>36b914d0bbdad6bf5ee88204ec44f3a5a9b47936</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-tags/src/components/TagGroup/renderTagGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
```
Comment (bsunderhus) [inline packages/react-components/react-tags/src/components/TagGroup/renderTagGroup.tsx:1]:
```suggestion
/** @jsxRuntime classic */
/** @jsx createElement */
import { createElement } from '@fluentui/react-jsx-runtime';
```
This is odd, I'm pretty sure I've added this to `create-component` tool 🤔

File: packages/react-components/react-tags/src/components/TagGroup/TagGroup.types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * TagGroup Props
+ */
+export type TagGroupProps = ComponentProps<TagGroupSlots> & {
+  size?: TagSize;
+};
+
+/**
+ * State used in rendering TagGroup
+ */
+export type TagGroupState = ComponentState<TagGroupSlots> & Required<Pick<TagGroupProps, 'size'>>;
```
Comment (bsunderhus) [inline packages/react-components/react-tags/src/components/TagGroup/TagGroup.types.ts:18]: Would be good to have a `TagGroupContextValues` type as similar to other components that present contextual value: https://github.com/microsoft/fluentui/blob/cd60ce9b960b7c107fee94ff41288540e917789a/packages/react-components/react-menu/src/components/Menu/Menu.types.ts#L171

File: packages/react-components/react-tags/src/components/TagGroup/useTagGroupContextValue.ts
Comment (bsunderhus) [inline packages/react-components/react-tags/src/components/TagGroup/useTagGroupContextValue.ts:1]: I'm being really picky here, I know but again, follow the patter in other components, this should be `useTagGroupContextValues`  

https://github.com/microsoft/fluentui/blob/cd60ce9b960b7c107fee94ff41288540e917789a/packages/react-components/react-menu/src/components/Menu/useMenuContextValues.ts

**Review by bsunderhus — CHANGES_REQUESTED** [review]
Please add the proper jsx comments on top of the render file otherwise `getSlotsNext` won't work properly.

Besides this LGTM.

File: packages/react-components/react-tags/src/components/TagGroup/renderTagGroup.tsx (hunk 1 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagGroup/renderTagGroup.tsx:1]: I created the component (last week?) and cherry-picked that commit. Then I had some problem with getSlots so I swapped to getSlotsNext. So probably back then you changes wasn't in master?

File: packages/react-components/react-tags/src/components/TagGroup/TagGroup.types.ts (hunk 2 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagGroup/TagGroup.types.ts:18]: Oh I see! yeah let me do this change

File: packages/react-components/react-tags/src/components/TagGroup/useTagGroupContextValues.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import type { TagGroupContextValues, TagGroupState } from './TagGroup.types';
+
+export function useTagGroupContextValues_unstable(state: TagGroupState): TagGroupContextValues {
+  const { size } = state;
+  return React.useMemo(() => ({ tagGroup: { size } }), [size]);
```
Comment (bsunderhus) [inline packages/react-components/react-tags/src/components/TagGroup/useTagGroupContextValues.ts:6]:
```suggestion
  return {tagGroup: React.useMemo(() => ({ size }), [size])};
```

nit: no need to context values, just the context value itself

**Review by bsunderhus — APPROVED** [review]
LGTM 💪🏻✅

**Pull request merged**